### PR TITLE
[COST-2643] pandas based rank limit

### PR DESCRIPTION
**Pull request by adberglund**
Fixes [COST-2643](https://issues.redhat.com/browse/COST-2643)

Changes proposed in this PR:
* This switches to using Pandas for the ranked list method and to compress the Others category when using filter[limit]
*

Testing Instructions:
1. Run `this command` ...
2. Ingest this data `with this other command` ...
3. Expect to see this result:
   ```
   GET http://localhost:8000/api/endpoint

   {json: response}
   ```
4.
5.

---
Additional Context:

### CODE REVIEW

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
297eafc0-2832-46aa-829a-4f58c8b8f82f
9dcc8392-278f-4662-bfc6-f15f30a4f598

**Comment by adberglund** [comment]
AWS 🚬 : https://ci.ext.devshift.net/job/project-koku-koku-pr-check/5442/

**Comment by adberglund** [comment]
AWS 🚬 : https://ci.ext.devshift.net/job/project-koku-koku-pr-check/5464/

**Comment by adberglund** [comment]
Azure 🚬 : https://ci.ext.devshift.net/job/project-koku-koku-pr-check/5468/ , test__ocp_on_azure_instance_reports is 💯 pass locally.

File: koku/api/report/ocp/provider_map.py [hunk 1]
```diff
@@ -1067,7 +1067,7 @@ def __init__(self, provider, report_type):
                             ),
                             "cost_units": Value("USD", output_field=CharField()),
                             "clusters": ArrayAgg(Coalesce("cluster_alias", "cluster_id"), distinct=True),
-                            "cost": Value(0, output_field=DecimalField()),
+                            # "cost": Value(0, output_field=DecimalField()),
```
Comment (abaiken) [inline koku/api/report/ocp/provider_map.py:1070]: Do we want to remove this rather than comment it out?

Comment (adberglund) [reply koku/api/report/ocp/provider_map.py:1070]:
```suggestion
```

**Review by abaiken — APPROVED** [review]
🌮 🎉  LGTM

**Pull request merged**